ui/src/features/Editor/components/OverlayUI/components/DebugActionBar/components/DebugStartPopover.tsx

```diff
@@ -36,7 +41,12 @@ const DebugStartPopover: React.FC<Props> = ({
         <div className="flex items-center justify-end">
           <Button
             variant="outline"
-            onClick={handleDebugRunStart}
+            onClick={
+              !customDebugRunWorkflowVariables ||
+              customDebugRunWorkflowVariables.length === 0
+                ? handleDebugRunStart
+                : onShowDebugWorkflowVariablesDialog
+            }
```

Comment:
It is cleaner to just have a single handleDebugRunStart here that itself decides whether to show the dialog or not. `DebugStartPopover` doesn't need to care